genie-agent-app/src/main/java/com/netflix/genie/GenieAgentApplication.java

```diff
@@ -60,6 +61,12 @@ private int run(final String[] args) {
         final SpringApplication app = new SpringApplication(GenieAgentApplication.class);
         // Disable parsing of command-line arguments into properties.
         app.setAddCommandLineProperties(false);
-        return SpringApplication.exit(app.run(args));
+
+        //TODO: workaround for https://jira.spring.io/browse/SPR-17416
+        // Spring chokes on argument '--' (a.k.a. bare double dash) conventionally used to separate options from
+        // operands. Perform a token replacement to avoid triggering an error in Spring argument parsing.
+        // Later the original token is restored before the actual Genie argument parsing.
+        final String[] editedArgs = Util.mangleBareDoubleDash(args);
```

Comment:
Instead of mangling and un-mangling did you consider just saving the original in some shared state location and later using it after Spring parsing? It's a minute optimization so not sure if it's worth it or not but would save you from doing all the replace logic twice

Reply:
Yeah, essentially using a global variable to store the args, then pass empty args to boot and then retrieve the original args from the global variable afterwards, from inside the app.
It just felt heavy-handed. `--` is really the only thing that Spring doesn't like, we may have reasons to let it process the rest.